modules/intentIqIdSystem.js

```diff
@@ -1,68 +1,68 @@
-/**
- * This module adds IntentIqId to the User ID module
- * The {@link module:modules/userId} module is required
- * @module modules/intentIqIdSystem
- * @requires module:modules/userId
- */
-
-import * as utils from '../src/utils.js'
-import {ajax} from '../src/ajax.js';
-import {submodule} from '../src/hook.js'
-
-const MODULE_NAME = 'intentIqId';
-
-/** @type {Submodule} */
-export const intentIqIdSubmodule = {
-  /**
-   * used to link submodule with config
-   * @type {string}
-   */
-  name: MODULE_NAME,
-  /**
-   * decode the stored id value for passing to bid requests
-   * @function
-   * @param {{ctrid:string}} value
-   * @returns {{intentIqId:string}}
-   */
-  decode(value) {
-    return (value && typeof value['ctrid'] === 'string') ? { 'intentIqId': value['ctrid'] } : undefined;
-  },
-  /**
-   * performs action to obtain id and return a value in the callback's response argument
-   * @function
-   * @param {SubmoduleParams} [configParams]
-   * @returns {IdResponse|undefined}
-   */
-  getId(configParams) {
-    if (!configParams || typeof configParams.partner !== 'number') {
-      utils.logError('User ID - intentIqId submodule requires a valid partner to be defined');
-      return;
-    }
-
-    // use protocol relative urls for http or https
-    const url = `https://api.intentiq.com/profiles_engine/ProfilesEngineServlet?at=39&mi=10&dpi=${configParams.partner}&pt=17&dpn=1`;
-    const resp = function (callback) {
-      const callbacks = {
-        success: response => {
-          let responseObj;
-          if (response) {
-            try {
-              responseObj = JSON.parse(response);
-            } catch (error) {
-              utils.logError(error);
-            }
-          }
-          callback(responseObj);
-        },
-        error: error => {
-          utils.logError(`${MODULE_NAME}: ID fetch encountered an error`, error);
-          callback();
-        }
-      };
-      ajax(url, callbacks, undefined, {method: 'GET', withCredentials: true});
-    };
-    return {callback: resp};
-  }
-};
-
-submodule('userId', intentIqIdSubmodule);
+/**
+ * This module adds IntentIqId to the User ID module
+ * The {@link module:modules/userId} module is required
+ * @module modules/intentIqIdSystem
+ * @requires module:modules/userId
+ */
+
+import * as utils from '../src/utils.js'
+import {ajax} from '../src/ajax.js';
+import {submodule} from '../src/hook.js'
+
+const MODULE_NAME = 'intentIqId';
+
+/** @type {Submodule} */
+export const intentIqIdSubmodule = {
+  /**
+   * used to link submodule with config
+   * @type {string}
+   */
+  name: MODULE_NAME,
+  /**
+   * decode the stored id value for passing to bid requests
+   * @function
+   * @param {{string}} value
+   * @returns {{intentIqId:string}}
+   */
+  decode(value) {
+    return (value != undefined && value != '') ? { 'intentIqId': value } : undefined;
```

Comment:
will this properly handle legacy values that are already stored in cache? it looks like for all users that hit a page with this new code, until a `getId` call is made again, you will lose the old value of the ID, which seems like it could be bad. Is that how this would work and is that intentional?